### PR TITLE
Add `dates_needed_for_prediction_range`

### PR DESCRIPTION
**Pull request by jeremyliweishih**

### CODE REVIEW

File: evalml/pipelines/time_series_pipeline_base.py [hunk 1]
```diff
@@ -317,3 +317,20 @@ def dates_needed_for_prediction(self, date):
...
+
+        Args:
+            start_date (pd.Timestamp): Start date of range to forecast in the future.
+            end_date (pd.Timestamp): End date of range to forecast in the future.
+
+        Returns:
+            dates_needed (tuple(pd.Timestamp)): Range of dates needed to forecast the given date.
+        """
+        if start_date > end_date:
+            raise ValueError("`start_date` must come before `end_date`.")
```
Comment (chukarsten) [inline evalml/pipelines/time_series_pipeline_base.py:332]: Just a nit, but we've had a `Raises:` section in the docstring.

**Review by chukarsten — APPROVED** [review]
I think just need to update the docstring!

File: docs/source/user_guide/timeseries.ipynb [hunk 2]
```diff
@@ -851,7 +851,128 @@
...
+   "source": [
+    "pl.predict(X_test_forecast, X_train=X_train_forecast, y_train=y_train_forecast)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": []
```
Comment (eccabay) [inline docs/source/user_guide/timeseries.ipynb:955]: This is an empty cell!

File: docs/source/user_guide/timeseries.ipynb [hunk 3]
```diff
@@ -851,7 +851,128 @@
...
+    "\n",
+    "dates = pl.dates_needed_for_prediction_range(forecast_start, forecast_end)\n",
+    "print(\"Dates needed:\")\n",
+    "print(f\"{dates[0].strftime('%Y-%m-%d %X')} to {dates[1].strftime('%Y-%m-%d %X')}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "tags": []
```
Comment (eccabay) [inline docs/source/user_guide/timeseries.ipynb:974]: This cell feels out of place to me, I'm not quite sure why it's here. Is it supposed to be somewhere else?

File: evalml/tests/pipeline_tests/test_time_series_pipeline.py [hunk 4]
```diff
@@ -1803,3 +1803,63 @@ def test_dates_needed_for_prediction(
...
+
+    prediction_start = pd.Timestamp("2022-02-01")
+    prediction_end = pd.Timestamp("2022-02-28")
+    beginning_date, end_date = pipeline.dates_needed_for_prediction_range(
+        prediction_start,
+        prediction_end,
+    )
+
+    assert beginning_date <= end_date
+    assert end_date < prediction_end
```
Comment (eccabay) [inline evalml/tests/pipeline_tests/test_time_series_pipeline.py:1857]: Just for future understanding's sake, could you add a comment here explaining what we're testing?

**Review by eccabay — CHANGES_REQUESTED** [review]
The code looks great! I think the addition to the docs needs a bit of clarification, it's hard to understand exactly what the goal is with it.

File: docs/source/user_guide/timeseries.ipynb [hunk 5]
```diff
@@ -851,7 +851,128 @@
   },
   {
    "cell_type": "markdown",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Forecasting into the future\n",
+    "\n",
+    "Our previous examples of shown using a pipeline to predict data we had at training time. However, we can also use EvalML time series pipelines to forecast dates into the future as long we provide data that meets the requirements of **Predicting on unseen data** as well.\n",
```
Comment (eccabay) [inline docs/source/user_guide/timeseries.ipynb:860]: Typo: "examples of shown" -> "examples have shown". Also maybe "to predict data we had at training time" -> "to predict *on* data we had at training time"

File: docs/source/user_guide/timeseries.ipynb (hunk 2 above)
Comment (eccabay) [reply docs/source/user_guide/timeseries.ipynb:955]: It's still here 👀

**Review by eccabay — APPROVED** [review]
Looks good! The documentation is much easier to understand. Just a couple typos, but otherwise let's 🚢

**Pull request merged**